include/dislocker/ssl_bindings.h.in

```diff
@@ -26,18 +26,13 @@
 /*
  * Here stand the bindings for polarssl SHA256/SHA2/SHA-2 function for dislocker
  */
-#include "@POLARSSL_INC_FOLDER@/config.h"
-#include "@POLARSSL_INC_FOLDER@/version.h"
+#include "@POLARSSL_INC_FOLDER@/build_info.h"
 #include "@POLARSSL_INC_FOLDER@/aes.h"
 
 // Function's name changed
 #if defined(MBEDTLS_SHA256_C)
 #  include "mbedtls/sha256.h"
-#  if MBEDTLS_VERSION_NUMBER >= 0x02070000
-#    define SHA256(input, len, output)         mbedtls_sha256_ret(input, len, output, 0)
-#  else
-#    define SHA256(input, len, output)         mbedtls_sha256(input, len, output, 0)
-#  endif /* POLARSSL_VERSION_NUMBER >= 0x02070000 */
+#  define SHA256(input, len, output)         mbedtls_sha256(input, len, output, 0)
```

Comment:
Why do you remove this check? I believe new versions of mbedtls to include only `mbedtls_sha256_ret` and not `mbedtls_sha256` (cf the deprecation notice [here](https://tls.mbed.org/api/sha256_8h.html#a25c94940f4217c4dd7ad6a4e7da61b47))

Reply:
@Aorimn

if MBEDTLS_VERSION_NUMBER >= 0x02070000 would include version 3.0 where mbedtls_sha256_ret is now mbedtls_sha256 again. 

https://github.com/Mbed-TLS/mbedtls/blob/development/docs/3.0-migration-guide.md#rename-mbedtls__ret-cryptography-functions-whose-deprecated-variants-have-been-removed

Reply:
Apologies to not respond earlier @Aorimn , but yes, @reallyallnamestaken is correct. This makes it work with the latest version of mbedTLS - using the most up to date versions of libraries is generally recommended for security anyway.

Reply:
If you do it like this it's a little bit nicer (and backwards compatible with version 2)

```
// Function's name changed
#if defined(MBEDTLS_SHA256_C)
#  include "mbedtls/sha256.h"
#  if MBEDTLS_VERSION_NUMBER >= 0x03000000
#    define SHA256(input, len, output)         mbedtls_sha256(input, len, output, 0)
#  elif MBEDTLS_VERSION_NUMBER >= 0x02070000
#    define SHA256(input, len, output)         mbedtls_sha256_ret(input, len, output, 0)
#  else
#    define SHA256(input, len, output)         mbedtls_sha256(input, len, output, 0)
#  endif /* POLARSSL_VERSION_NUMBER >= 0x02070000 */
#else /* defined(MBEDTLS_SHA256_C) */
```

And then similarly tackle it with an `elif` in `cmake/FindPolarSSL.cmake`